### ISSUE TITLE
[MPLUGIN-450] Make goal prefix mandatory by default

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-plugin-tools#767]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** opened **[MPLUGIN-450](https://issues.apache.org/jira/browse/MPLUGIN-450?redirect=false)** and commented

The goal prefix currently is not mandatory, and plugin uses heuristic to figure it out, if possible (xxx-maven-plugin or maven-xxx-plugin, latter for org.apache.maven plugins ONLY).

In general, goal prefix is optional, but "good to have", and usually is present.

Cases when is not present, is when heuristics fails (so plugin module is not named as "xxx-maven-plugin" or "maven-xxx-plugin") and user did not provide one either. The cases when prefix is not present is mostly unintentional, but maven-plugin-plugin leaves this without any remark or warning.

IMHO, the maven-plugin-plugin should either warn, or even fail the build in case when goal prefix is not present, but we may want to have some option to turn off this feature (for me unknown reasons, where it would be expected to NOT have goal prefix for a reason).

TL;DR Am pretty much sure that in most of the cases when plugin developer ends up with plugin without prefix, that it was unintentional (and causes surprise later, not to mention refactoring the module name or POM change to configure it), while there MAY be cases where goal prefix is not desired for some reason (ie. plugin not mentioned for manual/CLI invocation). Simply put, the out of the box defaults should enforce presence of it, while advanced users still can produce prefix-less plugins with some extra configuration. This would make things more explicit as well: even if plugin is named as xxx-maven-plugin, configuration would be clear I do not want prefix for it.

---

**Remote Links:**
- [GitHub Pull Request #239
](https://github.com/apache/maven-plugin-tools/pull/239)
- [GitHub Pull Request #240
](https://github.com/apache/maven-plugin-tools/pull/240)
